Library/Homebrew/cmd/install.rb

```diff
@@ -135,13 +135,41 @@ def install
           raise "No devel block is defined for #{f.full_name}"
         end
 
-        current = f if f.installed?
-        current ||= f.old_installed_formulae.first
+        if f.keg_only? && !f.installed_prefixes.empty? && f.opt_prefix.symlink? && !ARGV.force?
+          # keg-only install is only possible when no ohter version is
+          # optlinked, because silent install can break dependencies. Therefore
```

Comment:
`optlinked` -> `linked to opt`. Not sure I understand what "silent install" means.